addons/pinniped/config-controller/hack/run.sh

```diff
@@ -5,22 +5,141 @@
 
 set -euo pipefail
 
-MANIFESTS_DIR="./manifests"
-RBAC_FILE="${MANIFESTS_DIR}/role.yaml"
-DEPLOYMENT_FILE="${MANIFESTS_DIR}/deployment.yaml"
+YELLOW='\033[0;33m'
+DEFAULT='\033[0m'
+
 MY_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+TF_ROOT="${MY_DIR}/../../../.."
+
+APP_NAME="tanzu-auth"
+KAPP_CONTROLLER_NAME="${APP_NAME}-ctrl"
+
+# location for the package install
+MANAGEMENT_DIR="${TF_ROOT}/packages/management"
+TANZU_AUTH_DIR="${MANAGEMENT_DIR}/${APP_NAME}"
+TANZU_AUTH_BUNDLE_DIR="${TANZU_AUTH_DIR}/bundle"
+TANZU_AUTH_CONFIG_DIR="${TANZU_AUTH_BUNDLE_DIR}/config"
+
+TANZU_AUTH_PACKAGE_FILE="${TANZU_AUTH_DIR}/package.yaml"
+TANZU_AUTH_DEFAULT_VALUES_FILE="${TANZU_AUTH_BUNDLE_DIR}/values.yaml"
+
+# location to deploy the image
+REGISTRY_NAME="harbor-repo.vmware.com"
+REGISTRY_PROJECT="tkgiam"
+REGISTRY="${REGISTRY_NAME}/${REGISTRY_PROJECT}"
+
+# name of image to match that defined in /packages/management/tanzu-auth
+CONTROLLER_MANAGER_NAME="${APP_NAME}-controller-manager"
+CONTROLLER_MANAGER_PACKAGE_NAME="${CONTROLLER_MANAGER_NAME}-package"
+CONTROLLER_NAMESPACE_NAME="${APP_NAME}"
 
 # Always run from config-controller directory for reproducibility
 cd "${MY_DIR}/.."
 
-tag="dev"
-# tag="$(uuidgen)" # Uncomment to create random image every time
-image="harbor-repo.vmware.com/tkgiam/$(whoami)/pinniped-config-controller-manager:$tag"
+TAG="dev"
+# TAG="$(uuidgen)" # Uncomment to create random image every time
+
+# build the tanzu-auth-controller-manager image
+# --------------------------------------------------------
+CONTROLLER_IMAGE="${REGISTRY}/$(whoami)/${CONTROLLER_MANAGER_NAME}:${TAG}"
+echo -e "${YELLOW}building ${CONTROLLER_MANAGER_NAME} image and pushing to ${CONTROLLER_IMAGE}...${DEFAULT}"
+
+# push the tanzu-auth-controller-manager to the registry
+docker build -t "${CONTROLLER_IMAGE}" .
+docker push "${CONTROLLER_IMAGE}"
+
+
+# build the tanzu-auth-controller-manager-package image
+# --------------------------------------------------------
+PACKAGE_IMAGE="${REGISTRY}/$(whoami)/${CONTROLLER_MANAGER_PACKAGE_NAME}:${TAG}"
+echo -e "${YELLOW}building ${CONTROLLER_MANAGER_PACKAGE_NAME} image and pushing to ${PACKAGE_IMAGE}...${DEFAULT}"
 
+# generate new RBAC into /packages/management/tanzu-auth/bundle/config
+echo -e "${YELLOW}generating RBAC...${DEFAULT}"
 ./hack/generate.sh
-kubectl apply -f "${RBAC_FILE}"
 
-docker build -t "$image" .
-docker push "$image"
-ytt --data-value "image=$image" -f "${DEPLOYMENT_FILE}" | kbld -f - | kapp deploy -a pinniped-config-controller-manager -f - -y
-kubectl logs -n pinniped deploy/pinniped-config-controller-manager -f
+# the namespace must exist before the package can be deployed
+echo -e "${YELLOW}creating namespace ${APP_NAME}...${DEFAULT}"
+ytt --data-value "tanzuAuth.namespace=${APP_NAME}" --file "${TANZU_AUTH_CONFIG_DIR}/namespace.yaml" | kubectl apply -f -
+
+echo -e "${YELLOW}rebuilding image lock file with ytt and kbld...${DEFAULT}"
```

Comment:
ran `./hack.run.sh` and got an error: `kbld: Error: Writing images config: open /Users/akeesler/workspace/src/github.com/vmware-tanzu/tanzu-framework/addons/pinniped/config-controller/hack/../../../../packages/management/tanzu-auth/bundle/.imgpkg/images.yml: no such file or directory`

this resolved it: `mkdir packages/management/tanzu-auth/bundle/.imgpkg`

note - since you are injecting the image via `data.values.image`, then there is really no need for the `kbld` image relocation; feel free to leave it here for now just so we can get this merged

---

```akeesler@akeesler-a02:config-controller$ ./hack/run.sh
building tanzu-auth-controller-manager image and pushing to harbor-repo.vmware.com/tkgiam/akeesler/tanzu-auth-controller-manager:dev...
[+] Building 37.8s (16/16) FINISHED
...
 => => naming to harbor-repo.vmware.com/tkgiam/akeesler/tanzu-auth-controller-manager:dev                                                                                                                      0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
The push refers to repository [harbor-repo.vmware.com/tkgiam/akeesler/tanzu-auth-controller-manager]
0bfd65a7c0f4: Pushed
0b031aac6569: Mounted from tkgiam/akeesler/tkg-pinniped-post-deploy
dev: digest: sha256:b995dcf11bb542f7855d01663dc0a8b1d1bdcb129a67bbe11da685e269251f70 size: 739
building tanzu-auth-controller-manager-package image and pushing to harbor-repo.vmware.com/tkgiam/akeesler/tanzu-auth-controller-manager-package:dev...
generating RBAC...
make: Nothing to be done for `controller-gen'.
creating namespace tanzu-auth...
namespace/tanzu-auth created
rebuilding image lock file with ytt and kbld...
resolve | final: harbor-repo.vmware.com/tkgiam/akeesler/tanzu-auth-controller-manager:dev -> harbor-repo.vmware.com/tkgiam/akeesler/tanzu-auth-controller-manager@sha256:b995dcf11bb542f7855d01663dc0a8b1d1bdcb129a67bbe11da685e269251f70
kbld: Error: Writing images config: open /Users/akeesler/workspace/src/github.com/vmware-tanzu/tanzu-framework/addons/pinniped/config-controller/hack/../../../../packages/management/tanzu-auth/bundle/.imgpkg/images.yml: no such file or directory
```

Reply:
Ah good call, added `mkdir -p "${TANZU_AUTH_BUNDLE_DIR}/.imgpkg"`.